### ISSUE TITLE
transform_BLQ: M1 method appears not correctly implemented

### ISSUE THREAD

**Issue by schaedeli** [pharmpy/pharmpy#2085]
### Bug report

I am testing the transform_blq() function in pharmr version: 0.101.2. 
I am using a dataset that includes both a BLQ columen and an LLOQ column. For values BLQ (~10% of the observations) the DV has been set to LLOQ/2 in the dataset. 
For the method M1 I would expect that BLQ values should be ignored or masked somehow, resulting in a lower number of observations, however, this is not the case or I do not see how this has been implemented.
The following codes were tried for the M1 method: 
```
## 2 - transform the code for blq estimation M1 based on run0, a pharmr generated basic PK model
run1 <- run0 %>% set_name('run1') %>% transform_blq(method = "m1")
run1 %>% write_model(path="run1.mod")
## I do not see any statement to ignore BLQ values --> were the data removed from the dataset?
run1 <- read_model("run1.mod")
res1 <- fit(run1)
## same fit as run0 without any BLQ transformation, DVs treated as is 
```

Second code with lloq statement resulted in the same fit:

```
## M1 with explicit lloq statement
run2 <- run0 %>% set_name('run2') %>% transform_blq(method = "m1", lloq = 0.1)
run2 %>% write_model(path="run2.mod")
run2 <- read_model("run2.mod")
res2 <- fit(run2)
```

Here is the model code used as input, `run0` which was the same as the output model code:
```
$PROBLEM Start model
$INPUT C ID AMT TIME TYPE DV EVID NDAY NTAF PROT SITE DOSE AGE SEX RACE BHT BWT BIBW BBMI BBSA BSLN BASEAH50 SADDOSE SADAH50 BLQ BLQ2 LLOQ
$DATA dataset_filtered_blq2.csv IGNORE=@
$SUBROUTINES ADVAN2 TRANS2
$ABBR REPLACE ETA_CL=ETA(1)
$ABBR REPLACE ETA_VC=ETA(2)
$ABBR REPLACE ETA_MAT=ETA(3)
$PK
MAT = THETA(3)*EXP(ETA_MAT)
CL = THETA(1)*EXP(ETA_CL)
VC = THETA(2)*EXP(ETA_VC)
KA = 1/MAT
V = VC
$ERROR
IPRED = A(2)/VC
IF (IPRED.EQ.0) THEN
    IPREDADJ = 2.22500000000000E-16
ELSE
    IPREDADJ = IPRED
END IF
Y = IPRED + EPS(1)*IPREDADJ
$THETA  (0,6.0) ; POP_CL
$THETA  (0,600.0) ; POP_VC
$THETA  (0,0.1) ; POP_MAT
$OMEGA BLOCK(2)
0.1	; IIV_CL
0.01	; IIV_CL_IIV_VC
0.1	; IIV_VC
$OMEGA  0.1 ; IIV_MAT
$SIGMA  0.09 ; sigma
$ESTIMATION METHOD=COND INTER MAXEVAL=99999
$TABLE ID TIME DV CIPREDI CWRES FILE=mytab NOAPPEND NOPRINT

```
NONMEM code for `run1`:

```
$PROBLEM Start model with BLQ M1
$INPUT C ID AMT TIME TYPE DV EVID NDAY NTAF PROT SITE DOSE AGE SEX RACE BHT BWT BIBW BBMI BBSA BSLN BASEAH50 SADDOSE SADAH50 BLQ BLQ2 LLOQ
$DATA ../.datasets/run1.csv IGNORE=@
$SUBROUTINES ADVAN2 TRANS2
$ABBR REPLACE ETA_CL=ETA(1)
$ABBR REPLACE ETA_VC=ETA(2)
$ABBR REPLACE ETA_MAT=ETA(3)
$PK
MAT = THETA(3)*EXP(ETA_MAT)
CL = THETA(1)*EXP(ETA_CL)
VC = THETA(2)*EXP(ETA_VC)
KA = 1/MAT
V = VC
$ERROR
IPRED = A(2)/VC
IF (IPRED.EQ.0) THEN
    IPREDADJ = 2.22500000000000E-16
ELSE
    IPREDADJ = IPRED
END IF
Y = IPRED + EPS(1)*IPREDADJ
$THETA  (0,6.0) ; POP_CL
$THETA  (0,600.0) ; POP_VC
$THETA  (0,0.1) ; POP_MAT
$OMEGA BLOCK(2)
0.1	; IIV_CL
0.01	; IIV_CL_IIV_VC
0.1	; IIV_VC
$OMEGA  0.1 ; IIV_MAT
$SIGMA  0.09 ; sigma
$ESTIMATION METHOD=COND INTER MAXEVAL=99999
$TABLE ID TIME DV CIPREDI CWRES FILE=mytab NOAPPEND NOPRINT


```

And the head of the input dataset (original):

```
C,ID,AMT,TIME,TYPE,DV,EVID,NDAY,NTAF,PROT,SITE,DOSE,AGE,SEX,RACE,BHT,BWT,BIBW,BBMI,BBSA,BSLN,BASEAH50,SADDOSE,SADAH50,BLQ,BLQ2,LLOQ
0,3,10000,0,1,0,1,1,0,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,1,0,0,0,0.1
0,3,0,0.48,2,33.23,0,1,0.5,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,1,2,55.13,0,1,1,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,1.5,2,55.7,0,1,1.5,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,2,2,57.83,0,1,2,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,3.02,2,43.16,0,1,3,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,3.98,2,36.61,0,1,4,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,6,2,34.4,0,1,6,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,7.98,2,24.37,0,1,8,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,11.98,2,7.927,0,1,12,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,24.03,2,0.3525,0,2,24,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,36.03,2,0.1668,0,2,36,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,47.02,2,0.158,0,3,48,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,71.02,2,0.1491,0,4,72,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,167.02,2,0.1307,0,8,168,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,0,0,0.1
0,3,0,335.02,2,0.05,0,15,336,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,1,1,0.1
0,3,0,502.98,2,0.05,0,22,504,1,2093,10,33,0,1,173,67,68.65354331,22.4,1.799001558,39.95,60.5,0,0,1,2,0.1

```
Head of input dataset for run1:

```
C,ID,AMT,TIME,TYPE,DV,EVID,NDAY,NTAF,PROT,SITE,DOSE,AGE,SEX,RACE,BHT,BWT,BIBW,BBMI,BBSA,BSLN,BASEAH50,SADDOSE,SADAH50,BLQ,BLQ2,LLOQ
0.0,3,10000.0,0.0,1.0,0.0,1.0,1.0,0.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,1.0,0.0,0.0,0.0,0.1
0.0,3,0.0,0.48,2.0,33.23,0.0,1.0,0.5,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,1.0,2.0,55.13,0.0,1.0,1.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,1.5,2.0,55.7,0.0,1.0,1.5,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,2.0,2.0,57.83,0.0,1.0,2.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,3.02,2.0,43.16,0.0,1.0,3.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,3.98,2.0,36.61,0.0,1.0,4.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,6.0,2.0,34.4,0.0,1.0,6.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,7.98,2.0,24.37,0.0,1.0,8.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,11.98,2.0,7.927,0.0,1.0,12.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,24.03,2.0,0.3525,0.0,2.0,24.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,36.03,2.0,0.1668,0.0,2.0,36.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,47.02,2.0,0.158,0.0,3.0,48.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,71.02,2.0,0.1491,0.0,4.0,72.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,167.02,2.0,0.1307,0.0,8.0,168.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,0.0,0.0,0.1
0.0,3,0.0,335.02,2.0,0.05,0.0,15.0,336.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,1.0,1.0,0.1
0.0,3,0.0,502.98,2.0,0.05,0.0,22.0,504.0,1.0,2093.0,10.0,33.0,0.0,1.0,173.0,67.0,68.65354331,22.4,1.799001558,39.95,60.5,0.0,0.0,1.0,2.0,0.1

```

Here is also the datainfo file, in case it helps:
```
{"columns": [{"name": "C", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "ID", "type": "id", "scale": "nominal", "continuous": false, "categories": null, "unit": "1", "datatype": "int32", "drop": false, "descriptor": null}, {"name": "AMT", "type": "dose", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "TIME", "type": "idv", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "TYPE", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "DV", "type": "dv", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "EVID", "type": "event", "scale": "nominal", "continuous": false, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "NDAY", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "NTAF", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "PROT", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "SITE", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "DOSE", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "AGE", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "SEX", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "RACE", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "BHT", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "BWT", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "BIBW", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "BBMI", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "BBSA", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "BSLN", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "BASEAH50", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "SADDOSE", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "SADAH50", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "BLQ", "type": "blq", "scale": "nominal", "continuous": false, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "BLQ2", "type": "unknown", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}, {"name": "LLOQ", "type": "lloq", "scale": "ratio", "continuous": true, "categories": null, "unit": "1", "datatype": "float64", "drop": false, "descriptor": null}], "path": "run1.csv", "separator": ","}
```

**Comment by rikardn** [comment]
We would need to investigate this. The idea is that the `M1` method should simply remove the data from the dataset (at least for now).

**Comment by antoniaih** [comment]
This issue has been fixed in `ea0e437`.
The problem was that the `write_model` function was not writing the modified dataset. Thus, when reading the modified model the original dataset was used.
